crates/flipperzero/src/subghz/preset.rs

```diff
@@ -0,0 +1,64 @@
+use flipperzero_sys::{
+    FuriHalSubGhzPreset, FuriHalSubGhzPreset_FuriHalSubGhzPreset2FSKDev238Async,
+    FuriHalSubGhzPreset_FuriHalSubGhzPreset2FSKDev476Async,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetCustom,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetGFSK9_99KbAsync,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetIDLE,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetMSK99_97KbAsync,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetOok270Async,
+    FuriHalSubGhzPreset_FuriHalSubGhzPresetOok650Async,
+};
+
+#[derive(Clone, Debug)]
+pub enum SubGhzPreset<'a> {
+    /// default configuration
+    IDLE,
+    /// OOK, bandwidth 270kHz, asynchronous
+    Ook270Async,
+    /// OOK, bandwidth 650kHz, asynchronous
+    Ook650Async,
+    /// FM, deviation 2.380371 kHz, asynchronous
+    _2FSKDev238Async,
+    /// FM, deviation 47.60742 kHz, asynchronous
+    _2FSKDev476Async,
+    /// MSK, deviation 47.60742 kHz, 99.97Kb/s, asynchronous
+    MSK99_97KbAsync,
+    /// GFSK, deviation 19.042969 kHz, 9.996Kb/s, asynchronous
+    GFSK9_99KbAsync,
+    Custom(CustomSubGhzPreset<'a>),
+}
+
+impl<'a> SubGhzPreset<'a> {
+    pub fn into_furi_preset(&self) -> FuriHalSubGhzPreset {
```

Comment:
Not sure, but maybe its worth adding `impl From<&SubGhzPreset<'_>> for FuriHalSubGhzPreset>`

Also, `as_` prefix [seems to be][1] more valid for the function

[1]: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reply:
I am probably going to hold off for now on implementing this since [the docs](https://doc.rust-lang.org/std/convert/trait.From.html) mention that "The From trait is intended for perfect conversions," This function looses data when converting from the `Custom` variant.

I like the `as_` suggestion because it is something I always forget the semantics about (I should print it out, lol).
When I originally created some of the structs, I was not able to implement copy, but now I am  since the custom variant is `Custom(&[u8])`. Because of this, I am wondering if I should just change the function signature to take in `self` instead of a `&self` then maybe implying is should use `to_` ?